src/kpro_compress.erl

```diff
@@ -88,7 +88,8 @@ java_snappy_unpack_chunks(Chunks, Acc) ->
 
 do_compress(Name, IoData) ->
     Module = get_module(Name),
-    iodata(Module:compress(IoData)).
+    Data = maybe_convert_iodata_to_binary(Module, IoData),
```

Comment:
```suggestion
    Data = maybe_convert_iodata_to_binary(Module, iolist_to_binary(IoData)),
```

Reply:
no need for check binary, iolist_to_binary accepts binary() type

Reply:
Suggestion confused me a little, but as far as I understood the last comment, I deleted the `not is_binary(IoData)` guard.

Reply:
sorry, wrong suggestion.